src/features/QueueBrowser.js

```diff
@@ -0,0 +1,253 @@
+            browser.log('Connection failed to the message router: ' + sessionEvent.infoStr +
+                ' - check correct parameter values and connectivity!');
+        });
+        browser.session.on(solace.SessionEventCode.DISCONNECTED, function (sessionEvent) {
+            browser.log('Disconnected.');
+            browser.consuming = false;
+            if (browser.session !== null) {
+                browser.session.dispose();
+                browser.session = null;
+            }
+        });
+        // connect the session
+        try {
+            browser.session.connect();
+        } catch (error) {
+            browser.log(error.toString());
+        }
+    };
+
+    // Starts browsing from a queue on Solace PubSub+ Event Broker
+    browser.startBrowse = function () {
+        if (browser.session !== null) {
+            if (browser.consuming) {
+                browser.log('Already started browser for queue "' + browser.queueName +
+                    '" and ready to receive messages.');
+            } else {
+                browser.log('Starting browser for queue: ' + browser.queueName);
+                try {
+                    // Create a message browser
+                    browser.messageBrowser = browser.session.createQueueBrowser({
+                        queueDescriptor: { name: browser.queueName, type: solace.QueueType.QUEUE }
+                    });
+                    // Define message browser event listeners
+                    browser.messageBrowser.on(solace.QueueBrowserEventName.UP, function () {
+                        browser.consuming = true;
+                        browser.log('=== Ready to receive messages. ===');
+                    });
+                    browser.messageBrowser.on(solace.QueueBrowserEventName.CONNECT_FAILED_ERROR, function () {
+                        browser.consuming = false;
+                        browser.log('=== Error: the message browser could not bind to queue "' + browser.queueName +
+                            '" ===\n   Ensure this queue exists on the message router vpn');
+                        browser.exit();
+                    });
+                    browser.messageBrowser.on(solace.QueueBrowserEventName.DOWN, function () {
+                        browser.consuming = false;
+                        browser.log('=== The message browser is now down ===');
+                    });
+                    browser.messageBrowser.on(solace.QueueBrowserEventName.DOWN_ERROR, function () {
+                        browser.consuming = false;
+                        browser.log('=== An error happened, the message browser is down ===');
+                    });
+                    // Define message received event listener
+                    browser.messageBrowser.on(solace.QueueBrowserEventName.MESSAGE, function (message) {
+                        browser.log('Received message: ' + message.getGuaranteedMessageId());
```

Comment:
Is this actually receive message? It's not really receiving a message as much as its browsing the message, the message is not removed from the queue. Also we can add more information on the browsed message apart from just `getGuaranteedMessageId()`